### ISSUE TITLE
Crop on Save only

### ISSUE THREAD

**Issue by geetanshu-m** [joomla-projects/media-manager-improvement#554]
### Steps to reproduce the issue

1. Open Media-manager.
2. Go to edit tab for any image.
3. Change the dimensions of the overlay cropper
4. Change tab to rotate or resize.

### Expected result

Image should not be cropped until and unless save or apply button is being pressed.

### Actual result

Image is cropped instantly when the user left the crop tab.

**Comment by schnuti** [comment]
No. If I understand the issue and the workflow correct. Pressing the Save button saves the (so far) edited image to disk.  If you then press the Reset button you'll not work with the stored image. i.e. Cancel has a new meaning. You now have to save this state to store the "original" image. If Reset would fetch the stored image you couldn't revert changes at all.

**Comment by geetanshu-m** [comment]
My concerns are not the working of **save**, **reset** or the **cancel** button. What i am stating is when the selection is made in the crop section it is cropping instantly i.e. removing the rest of the part on the same time whenever the **crop** tab is left i.e. switching to the **resize** or **rotate** tab. I guess it should be like that.

**Comment by laoneo** [comment]
As every plugin does work in it's own canvas, it is almost impossible to move the state to the other tab. But you are right, some confirmation should happen when you switch the tab. Probably to ask if the changes should be reverted or applied and not just apply on tab switch.

**Comment by geetanshu-m** [comment]
>  some confirmation should happen when you switch the tab.Probably to ask if the changes should be reverted or applied and not just apply on tab switch.

Yes, this would be a good solution.

**Comment by schnuti** [comment]
I do not want to **click twice** in my normal workflow, only if I regret my action i.e I use the Reset button. As is. Clicking on a tab is a confirmation.

**Comment by schnuti** [comment]
In the code there are some ToDo functions including History and History redo. The future will tell us.
